pymongo/_version.py

```diff
@@ -15,16 +15,22 @@
 """Current version of PyMongo."""
 from __future__ import annotations
 
-from typing import Tuple, Union
+import re
+from typing import List, Tuple, Union
 
-version_tuple: Tuple[Union[int, str], ...] = (4, 8, 0, ".dev0")
+__version__ = "4.8.0.dev1"
 
-
-def get_version_string() -> str:
-    if isinstance(version_tuple[-1], str):
-        return ".".join(map(str, version_tuple[:-1])) + version_tuple[-1]
-    return ".".join(map(str, version_tuple))
+pattern = r"(?P<major>\d+).(?P<minor>\d+).(?P<patch>\d+)(?P<rest>.*)"
```

Comment:
Can we make this a function and add a few unit tests to make sure it works with a variety of versions? Eg "4.8.0.dev1", "4.8.0", "5.0", etc...

Reply:
Done.  We don't support "5.0", it has to have the patch version.

Reply:
I think not supporting "5.0" is a mistake. Our 4.0 release used "4.0" for example, same for "3.0". If we don't support "5.0" then get_version_tuple will silently return empty which is a bit odd.

Reply:
Fixed